### PR TITLE
🌱 Fixups after Makefile refactoring

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:

Did a bit of testing and we need some fixups.

Test release can be seen here: https://github.com/sbueringer/cluster-api-provider-vsphere/releases/tag/v1.8.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2051

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -529,7 +529,7 @@ dev-manifests: ## Builds dev manifests based on $(REGISTRY) and $(TAG) into the
 	$(MAKE) release-manifests STAGE=dev MANIFEST_DIR=$(OVERRIDES_DIR)
 
 .PHONY: manifest-modification
-manifest-modification: # Set the manifest images to $(REGISTRY)/$(IMAGE_NAME):$(RELEASE_TAG) and pull policy to $(PULL_POLICY)
+manifest-modification: $(BUILD_DIR) # Set the manifest images to $(REGISTRY)/$(IMAGE_NAME):$(RELEASE_TAG) and pull policy to $(PULL_POLICY)
```
Comment (sbueringer) [inline Makefile:532]: This didn't work in a GitHub action as the dir didn't exist before

File: Makefile [hunk 2]
```diff
@@ -571,7 +571,7 @@ release-alias-tag: ## Add the release alias tag to the last build tag
 	gcloud container images add-tag $(CONTROLLER_IMG):$(TAG) $(CONTROLLER_IMG):$(RELEASE_ALIAS_TAG)
 
 .PHONY: generate-release-notes
-generate-release-notes: $(RELEASE_NOTES_DIR)
+generate-release-notes: $(RELEASE_NOTES_DIR) $(RELEASE_NOTES)
```
Comment (sbueringer) [inline Makefile:574]: Required to build the binary

File: Makefile [hunk 3]
```diff
@@ -580,7 +580,7 @@ generate-release-notes: $(RELEASE_NOTES_DIR)
 
 .PHONY: promote-images
 promote-images: $(KPROMO)
-	$(KPROMO) pr --project cluster-api-provider-vsphere --tag $(RELEASE_TAG) --reviewers "$(IMAGE_REVIEWERS)" --fork $(USER_FORK) --image cluster-api-provider-vsphere
+	$(KPROMO) pr --project capi-vsphere --tag $(RELEASE_TAG) --reviewers "$(IMAGE_REVIEWERS)" --fork $(USER_FORK) --image cluster-api-vsphere-controller
```
Comment (sbueringer) [inline Makefile:583]: Fixups so it looks in the right project for our stuff 

(`gcr.io/k8s-staging-capi-vsphere/cluster-api-vsphere-controller:main`, project is prefixed with k8s-staging)

I think this command will only actually work once we did one release manually so we have a YAML with existing images in the image promotion repo (currently this yaml is just empty)

File: Makefile [hunk 4]
```diff
@@ -338,7 +338,7 @@ apidiff: $(GO_APIDIFF) ## Check for API differences
 ALL_VERIFY_CHECKS = boilerplate modules gen conversions
 
 .PHONY: verify
-verify: $(addprefix verify-,$(ALL_VERIFY_CHECKS)) ## Run all verify-* targets
+verify: $(addprefix verify-,$(ALL_VERIFY_CHECKS)) lint-markdown lint-shell ## Run all verify-* targets
```
Comment (sbueringer) [inline Makefile:341]: Adding those two linters so we can get rid of the separate jobs to simplify our job landscape

Comment (chrischdi) [reply Makefile:341]: Requires adding docker to the prowjob?!

Comment (sbueringer) [reply Makefile:341]: Yup PR will be opened in a bit

Comment (sbueringer) [reply Makefile:341]: @chrischdi https://github.com/kubernetes/test-infra/pull/30246

**Pull request merged**